coins/src/adapters/markets/goblin.ts

```diff
@@ -0,0 +1,84 @@
+import axios from "axios";
+import {
+  addToDBWritesList,
+  getTokenAndRedirectDataMap,
+} from "../utils/database";
+import { Write } from "../utils/dbInterfaces";
+
+const decimals = 8;
+const supplyQuery = 1000000;
+
+export async function goblin(timestamp: number = 0) {
+  if (timestamp != 0)
+    throw new Error("Goblin adapter only supports timestamp = 0");
+
+  const {
+    data: {
+      data: { vault },
+    },
+  } = await axios.post("https://api.hyperion.xyz/v1/graphql", {
+    query: `query fetchAllVaultQuery {
+            vault { tokenA  tokenB  vaultId  vaultName  }  
+        }`,
+  });
+
+  const underlyingAssets: string[] = [];
+  vault.map((v: any) => {
+    underlyingAssets.push(v.tokenA, v.tokenB);
+  });
+
+  const underlyingTokenData = await getTokenAndRedirectDataMap(
+    underlyingAssets,
+    "aptos",
+    timestamp
+  );
+
+  const writes: Write[] = [];
+  await Promise.all(
```

Comment:
can you replace this with PromisePool, this way, we dont make too many concurrent calls and get blocked by aptos rpc.

Also, this way, we can log the failed coins instead of failing the whole adapter if some calls fail